fullParseDice.py

```diff
@@ -0,0 +1,162 @@
+                for r in range(repetitions): #...we do the right side of the # as it's own expression, and return it.
+                    #KNOWN ISSUE: Ex: 3-2#1d20 will behave like 2#1d20. This can be mitigated by doing (3-2)#1d20 
+                    i.append(simplifyExpression(e[h+1:], dice, True, r+1)[0]) #puts all answers in an array.
+                return i #returns the array
+
+        elif e.count("d") > 0: #Checking for dice that need to be rolled. 
+            d = e.find("d") #Practically the exact code as above, but also moves the right side to find the edge of the right number as well.
+            leftbound = d
+            rightbound = d
+            while rightbound < len(e) -1:
+                if e[rightbound+1].isdigit():
+                    rightbound+=1
+                else:
+                    break
+
+            while leftbound > 0:
+                if e[leftbound-1].isdigit():
+                    leftbound-=1
+                else:
+                    break
+            rightbound+=1
+            rolled = parceDice(e[leftbound:rightbound], dice, diceInstance) #rolls dice and gets solution. Passes in dice and diceInstance so parceDice can record the details of the rolls.
+            e = replaceInstance(e,rolled,leftbound,rightbound) #replaces the d expression with what was rolled.
+
+        elif e.count("+") > 0: #Checking for Addition
+            plus = e.find("+")
+            bounds = findbounds(e, plus, "+") #The addition/subtraction version of the code above.
+            problem = e[bounds[0]:bounds[1]]
+            result = eval(problem) #Eval is vulnerable to code injection, but because of findbounds only numbers and +/- can get through. No issue.
+            if (problem[0] == "-" and str(result)[0] != "-"): #If something like "-30+60" occurred, without this we will 
+                result = "-"+str(result)
+            e = replaceInstance(e,str(result),bounds[0],bounds[1]) #replaces working expression with new change.
+
+        elif e.count("-") > 0: #Checking for Subtraction. This is almost identical to the above statement.
+            minus = e.find("-")
+            bounds = findbounds(e, minus, "-")
+            if bounds[0] == -1:
+                #This occurs if the only minus symbol found is actually a negative sign. We are fully simplified.
+                return e
+            
+            problem = e[bounds[0]:bounds[1]]
+            result = eval(problem) #by this point, there can't be anything left executable in the string.t)
+            e = replaceInstance(e,str(result),bounds[0],bounds[1])
+        try:
+            i = int(e)
+            break
+        except ValueError:
+            #Handle the exception
+            print("Simplified to:" + e)
+            if (eOld == e):
+                #Nothing has happened, something broke.
+                raise Exception("Invalid input string.")
+            eOld = e
+    return [i]
+
+def parceDice(expression, diceArray, instance):
```

Comment:
Again, make a docstring specifying what the function does. What are the inputs and what are the outputs. And if possible how it does it.